prover/crates/lib/snarkify_prover/src/types.rs

```diff
@@ -0,0 +1,84 @@
+use std::sync::Arc;
+
+use chrono::{DateTime, NaiveDateTime, Utc};
+use circuit_definitions::{
+    boojum::{cs::implementations::witness::WitnessVec, field::goldilocks::GoldilocksField},
+    circuit_definitions::recursion_layer::{
+        ZkSyncRecursionLayerVerificationKey, ZkSyncRecursionProof,
+    },
+};
+use serde::{Deserialize, Deserializer, Serialize};
+use zksync_prover_fri_types::CircuitWrapper;
+use zksync_prover_keystore::GoldilocksGpuProverSetupData;
+
+#[derive(Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum TaskState {
+    Pending,
+    Success,
+    Failure,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum ProofType {
+    Chunk,
+    Batch,
+    Bundle,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct CreateTaskRequest<Input> {
+    pub input: Input,
+    pub proof_type: ProofType,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct CompressionInput {
+    pub proof: ZkSyncRecursionProof,
+    pub scheduler_vk: ZkSyncRecursionLayerVerificationKey,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ProveInput {
+    pub circuit: CircuitWrapper,
+    pub witness_vector: WitnessVec<GoldilocksField>,
+    pub setup_data: Arc<GoldilocksGpuProverSetupData>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct CreateTaskResponse {}
```

Comment:
nit: `CreateTaskResponse` is actually `GetTaskResponse`, so you don't need it

Reply:
I renamed a few things:
- To prevent duplication of the crate name, remove `Snarkify` from the structs.
- Use `TaskReponse` to unify for Create, Get methods.